tests/plugins/test_pulp_pull.py

```diff
@@ -100,13 +101,29 @@ def custom_get_v1(self, url, headers, **kwargs):
     def custom_get_v2(self, url, headers, **kwargs):
         return self.config_response_config_v2
 
-    @pytest.mark.parametrize(('insecure', 'config_version'), [
-        (True, 'v1'),
-        (True, 'v2'),
-        (False, 'v1'),
-        (False, 'v2'),
+    @pytest.mark.parametrize('insecure', [(True, False)])
+    @pytest.mark.parametrize(('config_version', 'pulp_plugin', 'expected_version'), [
+        ('v1', [], []),
+        ('v1', [{'name': 'pulp_push'}], ['application/json']),
+        ('v1', [{'name': 'pulp_sync'}],
+         ['application/vnd.docker.distribution.manifest.v1+json']),
+        ('v1', [{'name': 'pulp_sync'}, {'name': 'pulp_push'}],
+         ['application/json',
+          'application/vnd.docker.distribution.manifest.v1+json']),
+        ('v2', [],
+         ['application/vnd.docker.distribution.manifest.v2+json']),
+        ('v2', [{'name': 'pulp_push'}],
+         ['application/json',
+          'application/vnd.docker.distribution.manifest.v2+json']),
+        ('v2', [{'name': 'pulp_sync'}],
+         ['application/vnd.docker.distribution.manifest.v1+json',
+          'application/vnd.docker.distribution.manifest.v2+json']),
+        ('v2', [{'name': 'pulp_sync'}, {'name': 'pulp_push'}],
+         ['application/json',
+          'application/vnd.docker.distribution.manifest.v1+json',
+          'application/vnd.docker.distribution.manifest.v2+json']),
     ])
-    def test_pull_first_time(self, insecure, config_version):
+    def test_pull_first_time(self, insecure, config_version, pulp_plugin, expected_version):
```

Comment:
Can we rename `config_version` to `schema_version`? Sorry, I know you didn't add it.